Update CalculatorController.java

PenguinPaladin opened the pull request.

PenguinPaladin committed:
Update CalculatorController.java

PenguinPaladin committed:
deleted:    HelloWorld/src/main/java/com/HelloWorld/HelloWorld/CalculatorService.java
	deleted:    HelloWorld/src/main/java/com/HelloWorld/HelloWorld/HelloWorldApplication.java

PenguinPaladin committed:
deleted:    HelloWorld/src/main/java/com/HelloWorld/HelloWorld/CalculatorController.java

PenguinPaladin committed:
deleted:    HelloWorld/src/main/java/com/HelloWorld/HelloWorld/CalculatorController.java

	modified:   HelloWorld/src/main/java/com/HelloWorld/HelloWorld/CalculatorController.java
	new file:   HelloWorld/src/main/java/com/HelloWorld/HelloWorld/CalculatorService.java
	new file:   HelloWorld/src/main/java/com/HelloWorld/HelloWorld/HelloWorldApplication.java

PenguinPaladin committed:
Merge branch 'main' of https://github.com/CoolNickname68/Education

dmitry-bizin commented on HelloWorld/src/main/java/com/HelloWorld/HelloWorld/CalculatorController.java:
```diff
@@ -4,8 +4,6 @@
-
-
 @RestController
 public class CalculatorController {
```
общий путь /calculator нужно вынести на уровень класса в аннотации RequestMapping, про это даже написано в задании

dmitry-bizin commented on HelloWorld/src/main/java/com/HelloWorld/HelloWorld/CalculatorService.java:
```diff
@@ -0,0 +1,20 @@
+@Service
+public class CalculatorService {
+
+    public String plus(int num1, int num2) {
```
математические методы сервиса не должны возвращать строки, они должны возвращать числа, потому что результат математической операции - это число, а не строка. В данном случае методы занимаются сразу 2 вещами - вычислением и формированием ответа (представления / View в паттерне MVC, см. шпаргалку). Так вот представлением, как и валидацией входных параметров, должен заниматься контроллер, это его задача. Задача сервиса - просто посчитать.

dmitry-bizin commented on HelloWorld/src/main/java/com/HelloWorld/HelloWorld/CalculatorService.java:
```diff
@@ -0,0 +1,20 @@
+        return num1 + " * " + num2 + " = " + (num1 * num2);
+    }
+    public String divide(int num1, int num2) {
+        return num1 + " / " + num2 + " = " + (num1 / num2);
```
нужна проверка знаменателя на 0 в контроллере

PenguinPaladin committed:
new commit

dmitry-bizin commented on HelloWorld/src/main/java/com/calculator/calculator/CalculatorController.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @GetMapping("/plus")
+    public int plus(@RequestParam int num1, @RequestParam int num2) {
```
методы контроллера должны возвращать строку в виде выражения. Методы сервиса должны возвращать числа

dmitry-bizin commented on HelloWorld/src/main/java/com/calculator/calculator/CalculatorController.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @GetMapping("/minus")
+    public int minus(@RequestParam int num1, @RequestParam int num2) {
```
методы контроллера должны возвращать строку в виде выражения. Методы сервиса должны возвращать числа

dmitry-bizin commented on HelloWorld/src/main/java/com/calculator/calculator/CalculatorController.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @GetMapping("/multiply")
+    public int multiply(@RequestParam int num1, @RequestParam int num2) {
```
методы контроллера должны возвращать строку в виде выражения. Методы сервиса должны возвращать числа

dmitry-bizin commented on HelloWorld/src/main/java/com/calculator/calculator/CalculatorController.java:
```diff
@@ -0,0 +1,48 @@
+        if (num2 == 0) {
+            return "Деление на 0 недопустимо";
+        } else {
+            return String.valueOf(calculatorService.divide(num1, num2));
```
методы контроллера должны возвращать строку в виде выражения. Методы сервиса должны возвращать числа

PenguinPaladin committed:
modified:   HelloWorld/src/main/java/com/calculator/calculator/CalculatorController.java
	modified:   HelloWorld/src/main/java/com/calculator/calculator/CalculatorService.java

PenguinPaladin committed:
modified:   HelloWorld/src/main/java/com/calculator/calculator/CalculatorController.java
	modified:   HelloWorld/src/main/java/com/calculator/calculator/CalculatorService.java

	deleted:    HelloWorld/src/main/java/com/calculator/calculator/CalculatorApp.java
	modified:   HelloWorld/src/main/java/com/calculator/calculator/CalculatorController.java
	modified:   HelloWorld/src/main/java/com/calculator/calculator/CalculatorService.java
	new file:   HelloWorld/src/main/java/com/calculator/calculator/CredentialsValidator.java

PenguinPaladin committed:
Merge branch 'main' of https://github.com/CoolNickname68/Education into main2

PenguinPaladin committed:
modified:   HelloWorld/src/main/java/com/calculator/calculator/CredentialsValidator.java

PenguinPaladin committed:
deleted:    HelloWorld/src/main/java/com/calculator/calculator/CredentialsValidator.java
	new file:   HelloWorld/src/main/java/com/calculator/calculator/Employee.java
	new file:   HelloWorld/src/main/java/com/employee/employee/Application.java
	new file:   HelloWorld/src/main/java/com/employee/employee/Employee.java
	new file:   HelloWorld/src/main/java/com/employee/employee/EmployeeController.java
	new file:   HelloWorld/src/main/java/com/employee/employee/EmployeeService.java
	new file:   HelloWorld/src/main/java/com/employee/employee/exception/EmployeeAlreadyAddedException.java
	new file:   HelloWorld/src/main/java/com/employee/employee/exception/EmployeeNotFoundException.java
	new file:   HelloWorld/src/main/java/com/employee/employee/exception/EmployeeStorageIsFullException.java

PenguinPaladin committed:
new file:   HelloWorld/src/main/java/com/employee/employee/DepartmentController.java
	modified:   HelloWorld/src/main/java/com/employee/employee/Employee.java
	modified:   HelloWorld/src/main/java/com/employee/employee/EmployeeController.java
	modified:   HelloWorld/src/main/java/com/employee/employee/EmployeeService.java

dmitry-bizin commented on HelloWorld/src/main/java/com/calculator/calculator/Employee.java:
```diff
@@ -0,0 +1,13 @@
+        this.firstName = firstName;
+        this.lastName = lastName;
+    }
+
```
а где конструктор по умолчанию, геттеры и сеттеры? Jackson (библиотека сериализации в JSON, которую использует Spring, не сможет сделать json из такого объекта без конструктора по умолчанию, а также геттеров и сеттеров)

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/DepartmentController.java:
```diff
@@ -0,0 +1,48 @@
+        this.employeeService = employeeService;
+    }
+
+    @GetMapping("/departments/max-salary")
```
общий путь departments лучше вынести на уровень класса

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/DepartmentController.java:
```diff
@@ -0,0 +1,48 @@
+    @GetMapping("/departments/max-salary")
+    public ResponseEntity<Employee> getEmployeeWithMaxSalary(@RequestParam int departmentId) {
+        Employee employee = employeeService.getEmployeeWithMaxSalaryByDepartment(departmentId);
+        return ResponseEntity.ok(employee);
```
нет никакого смысла в оборачивании ответа в ResponseEntity с кодом 200 (ok метод). Если просто вернуть объект, то Spring по умолчанию добавит код 200, поэтому лишний код (оборачивание в ResponseEntity) лучше не писать

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/DepartmentController.java:
```diff
@@ -0,0 +1,48 @@
+    @GetMapping("/departments/min-salary")
+    public ResponseEntity<Employee> getEmployeeWithMinSalary(@RequestParam int departmentId) {
+        Employee employee = employeeService.getEmployeeWithMinSalaryByDepartment(departmentId);
+        return ResponseEntity.ok(employee);
```
нет никакого смысла в оборачивании ответа в ResponseEntity с кодом 200 (ok метод). Если просто вернуть объект, то Spring по умолчанию добавит код 200, поэтому лишний код (оборачивание в ResponseEntity) лучше не писать

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/DepartmentController.java:
```diff
@@ -0,0 +1,48 @@
+    public ResponseEntity<List<Employee>> getAllEmployeesByDepartment(@RequestParam(required = false) Integer departmentId) {
+        if (departmentId != null) {
+            List<Employee> employees = employeeService.getAllEmployeesByDepartment(departmentId);
+            return ResponseEntity.ok(employees);
```
нет никакого смысла в оборачивании ответа в ResponseEntity с кодом 200 (ok метод). Если просто вернуть объект, то Spring по умолчанию добавит код 200, поэтому лишний код (оборачивание в ResponseEntity) лучше не писать

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/DepartmentController.java:
```diff
@@ -0,0 +1,48 @@
+            return ResponseEntity.ok(employees);
+        } else {
+            List<Employee> allEmployees = employeeService.getAllEmployees();
+            return ResponseEntity.ok(allEmployees);
```
нет никакого смысла в оборачивании ответа в ResponseEntity с кодом 200 (ok метод). Если просто вернуть объект, то Spring по умолчанию добавит код 200, поэтому лишний код (оборачивание в ResponseEntity) лучше не писать

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/DepartmentController.java:
```diff
@@ -0,0 +1,48 @@
+    @GetMapping("/departments/all-grouped")
+    public ResponseEntity<Map<String, List<Employee>>> getAllEmployeesGroupedByDepartment() {
+        Map<String, List<Employee>> employeesGroupedByDepartment = employeeService.getAllEmployeesGroupedByDepartment();
+        return ResponseEntity.ok(employeesGroupedByDepartment);
```
нет никакого смысла в оборачивании ответа в ResponseEntity с кодом 200 (ok метод). Если просто вернуть объект, то Spring по умолчанию добавит код 200, поэтому лишний код (оборачивание в ResponseEntity) лучше не писать

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/DepartmentController.java:
```diff
@@ -0,0 +1,48 @@
+        return ResponseEntity.ok(employee);
+    }
+
+    @GetMapping("/departments/min-salary")
```
общий путь departments лучше вынести на уровень класса

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/DepartmentController.java:
```diff
@@ -0,0 +1,48 @@
+        return ResponseEntity.ok(employee);
+    }
+
+    @GetMapping("/departments/all")
```
общий путь departments лучше вынести на уровень класса

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/DepartmentController.java:
```diff
@@ -0,0 +1,48 @@
+        }
+    }
+
+    @GetMapping("/departments/all-grouped")
```
общий путь departments лучше вынести на уровень класса

dmitry-bizin commented on HelloWorld/src/main/java/com/calculator/calculator/Employee.java:
```diff
@@ -0,0 +1,13 @@
+package com.calculator.calculator;
+
+public class Employee {
```
этот класс в итоге не используется, зачем тогда он?

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/Employee.java:
```diff
@@ -0,0 +1,88 @@
+    private String firstName;
+    private String lastName;
+    private double salary;
+    private String department;
+    private int departmentId; // новое поле для идентификатора отдела
```
должно быть 1 поле - id отдела или название, потому что суть у них одна и та же и заполняется только строковое

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/EmployeeController.java:
```diff
@@ -0,0 +1,71 @@
+    @GetMapping("/all")
+    public ResponseEntity<List<Employee>> getAllEmployees() {
+        List<Employee> employees = employeeService.getAllEmployees();
+        return ResponseEntity.ok(employees);
```
здесь тоже нет смысла в ResponseEntity

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/EmployeeService.java:
```diff
@@ -0,0 +1,88 @@
+
+    public Employee removeEmployee(String firstName, String lastName) {
+        Employee employeeToRemove = findEmployee(firstName, lastName);
+        if (employeeToRemove == null) {
```
null никогда не вернётся (см. внимательнее код метода `findEmployee`), поэтому проверка смысла не имеет, она всегда будет ложна

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/DepartmentController.java:
```diff
@@ -0,0 +1,48 @@
+@RestController
+public class DepartmentController {
+    
+    private final EmployeeService employeeService;
```
здесь надо использовать отдельный сервис - DepartmentService - с методами по работе с отделами

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/EmployeeService.java:
```diff
@@ -0,0 +1,88 @@
+        return employeeList;
+    }
+
+    public Employee getEmployeeWithMaxSalaryByDepartment(int departmentId) {
```
этот метод должен быть в DepartmentService

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/EmployeeService.java:
```diff
@@ -0,0 +1,88 @@
+                .orElseThrow(() -> new RuntimeException("No employees found for department: " + departmentId));
+    }
+
+    public Employee getEmployeeWithMinSalaryByDepartment(int departmentId) {
```
этот метод должен быть в DepartmentService

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/EmployeeService.java:
```diff
@@ -0,0 +1,88 @@
+                .orElseThrow(() -> new RuntimeException("No employees found for department: " + departmentId));
+    }
+
+    public List<Employee> getAllEmployeesByDepartment(int departmentId) {
```
этот метод должен быть в DepartmentService

dmitry-bizin commented on HelloWorld/src/main/java/com/employee/employee/EmployeeService.java:
```diff
@@ -0,0 +1,88 @@
+                .collect(Collectors.toList());
+    }
+
+    public Map<String, List<Employee>> getAllEmployeesGroupedByDepartment() {
```
этот метод должен быть в DepartmentService

PenguinPaladin committed:
new file:   HelloWorld/src/main/java/com/collection/collection/collection.java

PenguinPaladin committed:
modified:   HelloWorld/pom.xml
	modified:   HelloWorld/src/main/java/com/employee/employee/EmployeeController.java
	new file:   HelloWorld/src/main/java/com/employee/employee/EmployeeDTO.java
	modified:   HelloWorld/src/main/java/com/employee/employee/EmployeeService.java
